app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    counter = Post.where(author_id:).count
+    user = User.find_by(id: author_id)
+    user.update(posts_counter: counter)
+  end
+
+  def recent_comments
+    Comment.where(post_id: id).limit(5).order(created_at: :desc)
+  end
```

Comment:
- [ ] A more effective query would be:
```rb
def recent_comments
  comments.order('created_at Desc').limit(5)
end
```
Remember that the comments have been already associated with the posts.

Reply:
✅ Resolved.